src/pudl/transform/eia861.py

```diff
@@ -603,6 +606,56 @@ def _early_transform(df):
     return df
 
 
+def _compare_totals(data_cols, idx_cols, class_type, df_name):
+    """Compare reported totals with sum of component columns.
+
+    Args:
+        data_cols (pd.DataFrame): A DataFrame containing only the columns with
+            normalized information.
+        idx_cols (list): A list of the primary keys for the given denormalized
+            DataFrame.
+        class_type (str): The name (either 'customer_class' or 'fuel_class') of
+            the column for which you'd like to compare totals to components.
+        df_name (str): The name of the dataframe.
+    """
+    # Convert column dtypes so that numeric cols can be adequately summed
+    data_cols = pudl.helpers.convert_cols_dtypes(data_cols, 'eia')
+    # Drop data cols that are non numeric (preserve primary keys)
+    data_cols = (
+        data_cols.set_index(idx_cols + [class_type])
+        .select_dtypes('float', 'Int64')
```

Comment:
this is neat! I had never seen `select_dtypes` but this makes a lot of sense in this context. My one suggestion here is to use the more generic `np.number` or just `number`. Because this might miss the `int64` or `Int32` or any of the other nullable or non-nullable ints .